Structural/Decorator/JsonRenderer.php

```diff
@@ -8,4 +8,16 @@ public function renderData(): string
     {
         return json_encode($this->wrapped->renderData());
     }
+
+    /**
+     * Renders data as JSON based on options passed.
+     * In terms of decorator patters, it represents additional extended behaviour.
+     *
+     * @param int $options Options to be used to encode data into JSON.
+     * @return string Rendered output
+     */
+    public function renderDataWithOptions($options = 0): string
```

Comment:
A good example of chaining decorators https://sourcemaking.com/design_patterns/decorator
But maybe in the current Json/Xml example chaining is not useful at all. You will render Xml, Json or another one like Csv for example.